services/src/main/java/org/keycloak/authentication/authenticators/browser/AbstractUsernameFormAuthenticator.java

```diff
@@ -209,30 +201,29 @@ public boolean validatePassword(AuthenticationFlowContext context, UserModel use
 
     public boolean validatePassword(AuthenticationFlowContext context, UserModel user, MultivaluedMap<String, String> inputData, boolean clearUser) {
         String password = inputData.getFirst(CredentialRepresentation.PASSWORD);
+
         if (password == null || password.isEmpty()) {
             return badPasswordHandler(context, user, clearUser,true);
         }
-
-        if (isDisabledByBruteForce(context, user)) return false;
-
-        if (password != null && !password.isEmpty() && context.getSession().userCredentialManager().isValid(context.getRealm(), user, UserCredentialModel.password(password))) {
+        if (isDisabledByBruteForce(context, user)) {
+            return false;
+        }
+        if (context.getSession().userCredentialManager().isValid(context.getRealm(), user, UserCredentialModel.password(password))) {
```

Comment:
Not sure if this actually changes any behaviour, but it would be safer and cleaner to just keep the same method call to validate the password
```suggestion
        if (user.credentialManager().isValid(UserCredentialModel.password(password))) {
```